### PR TITLE
Profiles: Add Favorite Pokémon

### PR DESCRIPTION
**Pull request by DeathlyPlays**

### CODE REVIEW

File: wavelength-plugins/profile.js [hunk 1]
```diff
@@ -274,6 +274,41 @@ exports.commands = {
 			);
 		},
 	},
+
+	pokemon: {
+		add: "set",
+		set: function (target, room, user) {
+			let dex = Dex.data.Pokedex.species;
+			if (!target) return this.parse("/pokemonhelp");
+			let pkmn = target.toLowerCase().replace(' ', '');
```
Comment (HoeenCoder) [inline wavelength-plugins/profile.js]: toId()

File: wavelength-plugins/profile.js [hunk 2]
```diff
@@ -274,6 +274,41 @@ exports.commands = {
...
+			if (!Dex.data.Pokedex[pkmn]) {
+				return this.errorReply('Not a Pokemon. Check your spelling?');
+			}
+			if (Db.pokemon.has(target)) return this.errorReply("This Pokemon is already listed as your favorite Pokemon on your profile.");
```
Comment (HoeenCoder) [inline wavelength-plugins/profile.js]: Feel free to just overide the existing one unless it matches.

File: wavelength-plugins/profile.js [hunk 3]
```diff
@@ -274,6 +274,41 @@ exports.commands = {
...
+			Db.pokemon.set(toId(user), target);
+			this.sendReply("You have successfully set your Pokemon onto your profile.");
+		},
+
+		del: "delete",
+		remove: "delete",
+		delete: function (target, room, user) {
+			if (!target) {
```
Comment (HoeenCoder) [inline wavelength-plugins/profile.js]: why this?

File: wavelength-plugins/profile.js [hunk 4]
```diff
@@ -274,6 +274,41 @@ exports.commands = {
...
+		add: "set",
+		set: function (target, room, user) {
+			let dex = Dex.data.Pokedex.species;
+			if (!target) return this.parse("/pokemonhelp");
+			let pkmn = target.toLowerCase().replace(' ', '');
+			if (!Dex.data.Pokedex[pkmn]) {
+				return this.errorReply('Not a Pokemon. Check your spelling?');
+			}
+			if (Db.pokemon.has(target)) return this.errorReply("This Pokemon is already listed as your favorite Pokemon on your profile.");
+			Db.pokemon.set(toId(user), target);
```
Comment (HoeenCoder) [inline wavelength-plugins/profile.js]: Set the exact species name

File: wavelength-plugins/profile.js (hunk 2 above)
Comment (DeathlyPlays) [reply wavelength-plugins/profile.js]: It currently does anyways lol this code is irrelevant tbh

Comment (DeathlyPlays) [reply wavelength-plugins/profile.js]: Will remove

File: wavelength-plugins/profile.js (hunk 3 above)
Comment (DeathlyPlays) [reply wavelength-plugins/profile.js]: Tbh idk >_>

File: wavelength-plugins/profile.js (hunk 4 above)
Comment (DeathlyPlays) [reply wavelength-plugins/profile.js]: So dex?

File: wavelength-plugins/profile.js [hunk 5]
```diff
@@ -274,6 +274,38 @@ exports.commands = {
 			);
 		},
 	},
+
+	pokemon: {
+		add: "set",
+		set: function (target, room, user) {
+			let dex = Dex.data.Pokedex.species;
```
Comment (HoeenCoder) [inline wavelength-plugins/profile.js]: This is pointless and undefined

File: wavelength-plugins/profile.js [hunk 6]
```diff
@@ -274,6 +274,38 @@ exports.commands = {
...
+			if (!target) return this.parse("/pokemonhelp");
+			let pkmn = toId(target);
+			if (!Dex.data.Pokedex[pkmn]) {
+				return this.errorReply('Not a Pokemon. Check your spelling?');
+			}
+			Db.pokemon.set(toId(user), dex);
```
Comment (HoeenCoder) [inline wavelength-plugins/profile.js]: This should be set to the pokemon not something else

File: wavelength-plugins/profile.js [hunk 7]
```diff
@@ -274,6 +274,35 @@ exports.commands = {
...
+			if (!pkmn.exists) return this.errorReply('Not a Pokemon. Check your spelling?');
+			Db.pokemon.set(user.userid, pkmn.species);
+			return this.sendReply("You have successfully set your Pokemon onto your profile.");
+		},
+
+		del: "delete",
+		remove: "delete",
+		delete: function (target, room, user) {
+			if (!Db.pokemon.has(toId(user))) return this.errorReply("Your favorite Pokemon hasn't been set.");
+			Db.pokemon.remove(toId(user));
```
Comment (HoeenCoder) [inline wavelength-plugins/profile.js]: user.userid not toId(user)

Comment (DeathlyPlays) [reply wavelength-plugins/profile.js]: Ok

**Pull request merged**